search-sentiments-map/public/scripts/modal-script.js

```diff
@@ -71,10 +71,11 @@ function toggleDisplay(id) {
 }
 
 /** 
- * Displays the top results in a country for current search trend on modal. 
+ * Displays the top results in a country for current search trend on modal
+ * and show positive/negative words depending on the sign of the sentiment score.  
  * @param {string} countryCode Two letter country code for selected country.
  */
-function displayTopResultsForCurrentTrend(countryCode) {
+async function displayTopResultsForCurrentTrend(countryCode) {
```

Comment:
This function seems too long. Consider dividing it up.

Reply:
It is 60 lines long, but I think since all the code is relevant to the function title, and we do have other 60 lines long functions. I think it'd be more clear to just have the same code in there bc it is relevant to the function instead of sending it to another smaller function. What do others think? @hayleejane3 @carmenBen @lawrencezjh

Reply:
Which other function is 60 lines long? This function has too many layers of logic and I think you can put a coherent subsection in a different function.

Reply:
The name is completely fine. I've seen functions whose names don't even fit on a single line!
Descriptiveness is good in this case.

Reply:
@hayleejane3 We are talking about the length of the actual function, not the name

Reply:
retrieveUserSearchResultFromDatastore in search.js is 50+ lines, I just think the layers of logic makes most sense to be left in this since this function is supposed to be updating resultElement.innerHTML.

Reply:
made it shorter with the new logic

Reply:
my bad